examples/js/loaders/GLTFLoader.js

```diff
@@ -12,368 +16,1836 @@ THREE.GLTFLoader.prototype = {
 
 		var scope = this;
 
+		var baseUrl = this.baseUrl && ( typeof this.baseUrl === "string" ) ? this.baseUrl : THREE.Loader.prototype.extractUrlBase( url );
+
 		var loader = new THREE.XHRLoader( scope.manager );
-		loader.load( url, function ( text ) {
+		loader.load( url, function( text ) {
 
-			onLoad( scope.parse( JSON.parse( text ) ) );
+			scope.parse( JSON.parse( text ), onLoad, baseUrl );
 
 		}, onProgress, onError );
 
 	},
 
-	setCrossOrigin: function ( value ) {
+	setCrossOrigin: function( value ) {
 
 		this.crossOrigin = value;
 
 	},
 
-	parse: function ( json ) {
+	setBaseUrl: function( value ) {
+
+		this.baseUrl = value;
+
+	},
+
+	parse: function( json, callback, baseUrl ) {
+
+		console.time( 'GLTFLoader' );
+
+		var glTFParser = new GLTFParser( json, {
+			baseUrl: baseUrl || this.baseUrl,
+			crossOrigin: !!this.crossOrigin
+		});
+
+		glTFParser.parse( function( library ) {
+
+			console.timeEnd( 'GLTFLoader' );
+
+			var scene = library.scenes[ json.scene ];
+
+			var cameras = [];
+			_each( library.cameras, function( camera ) {
+
+				cameras.push( camera );
+
+			});
+
+			var animations = [];
+			_each( library.animations, function( animation ) {
+
+				animations.push( animation );
+
+			});
+
+			var glTF = {
+
+				scene		:	scene,
+				cameras		:	cameras,
+				animations	:	animations
+
+			};
+
+			callback( glTF );
+
+		});
+
+		// Developers should use `callback` argument for async notification on
+		// completion to prevent side effects.
+		// Function return is kept only for backward-compatability purposes.
+		return {
+			get scene() {
+
+				console.error( "synchronous glTF object access is deprecated. " +
+						"Instead use 'callback' argument in function call " +
+						"to access asynchronous glTF object." );
+				return library.scenes[ json.scene ];
+
+			},
+			set scene( value ) {
+
+				library.scenes[ json.scene ] = value;
+
+			}
+
+		};
+
+	}
+
+};
+
+/* GLTFREGISTRY */
+
+var GLTFRegistry = function() {
+
+	var objects = [];
+
+	return	{
+		add : function( object ) {
 
-		function stringToArrayBuffer( string ) {
+			objects.push( object );
 
-			var bytes = atob( string );
-			var buffer = new ArrayBuffer( bytes.length );
-			var bufferView = new Uint8Array( buffer );
+		},
 
-			for ( var i = 0; i < bytes.length; i ++ ) {
+		remove: function( object ) {
 
-				bufferView[ i ] = bytes.charCodeAt( i );
+			var i = objects.indexOf( object );
+
+			if ( i !== - 1 ) {
+
+				objects.splice( i, 1 );
 
 			}
 
-			return buffer;
+		},
+
+		removeAll: function() {
+
+			objects = []
+
+		},
+
+		update : function( scene, camera ) {
+
+			_each( objects, function( object ) {
+
+				object.update( scene, camera );
+
+			});
 
 		}
+	};
+};
 
-		console.time( 'GLTFLoader' );
+/* GLTFSHADERS */
 
-		var library = {
-			buffers: {},
-			bufferViews: {},
-			accessors: {},
-			textures: {},
-			materials: {},
-			meshes: {},
-			nodes: {},
-			scenes: {}
-		};
+THREE.GLTFShaders = new GLTFRegistry();
```

Comment:
Good idea and updated in e0e7dbf9fefa8ee6a01e580ca0801fd2c9bf0f40.

`THREE.GLTFShaders` -> `THREE.GLTFLoader.Shaders`
`THREE.GLTFAnimator` -> `THREE.GLTFLoader.Animations`